### PR TITLE
Fixes dragons not dropping their consumed mobs on despawn

### LINKED ISSUE #71536: Unequipped memento mori dusts Space Dragon and contents when the dragon dies.
Server: /tg/Station Manuel
[Round ID]: #195076

BYOND Version: 514.1589
Server revision compiled on: 
Master commit: [b52b4f8dd7e0c8e1db250eaa934e6338aa60c29f](https://www.github.com/tgstation/tgstation/commit/b52b4f8dd7e0c8e1db250eaa934e6338aa60c29f)
The following pull requests are currently test merged:
[#71520: 'fixes & increases SSinput click delay threshold & moves it and average_click_delay to deciseconds' by Kylerace at commit 73347fa0f9](https://www.github.com/tgstation/tgstation/pull/71520)
[#71474: 'Revert "Unit tests radio, saymode, and language prefix keys"' by MrStonedOne at commit 19d0ba1b7e](https://www.github.com/tgstation/tgstation/pull/71474)
[#71527: 'Partially reverts moving up/down whilst in a movable from #70431' by Watermelon914 at commit bb1ccc9005](https://www.github.com/tgstation/tgstation/pull/71527)

Local commit: eeaa69a7add5c467b383f92428982ae8fb8d5ef8
TGS version: /tg/station 13 Server v3.2.6.2
DMAPI version: 3.2.1.3

Current Informational Settings:
Protect Authority Roles From Traitor: 1
Protect Assistant Role From Traitor: 0
Enforce Human Authority: 1
Allow Latejoin Antagonists: 1

## Reproduction:

Step 1: Become a space dragon.
Step 2: Eat a bunch of people, or at the very least a person with a Memento Mori stored in a bag.
Step 3: Get killed due to not placing enough rifts in time.

As a result of these three steps, we found that every individual consumed by the space dragon will get dusted when the dragon dies. The unused memento mori as well as any trading cards in the character inventories will be the only remaining items. Another interesting result is that any Wisps equipped will remain active even after the original owner has been dusted by the death of the dragon.

image

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

This was fixed a while back, but was accidentally reverted / regressed due to a refactor

Basically, the dragon needs to be killed before being deleted, being killed will drop all of their stuff and handle "dragon is dead" events, then it can be fully deleted and removed as expected

Unit tests it, since this is a regression

Fixes #71536

## Why It's Good For The Game

Having a lot of mobs deleted is kinda really bad

## Changelog

:cl: Melbert
fix: Fixes dragon despawn deleting all the people they consumed
/:cl:

### CODE REVIEW

File: code/modules/antagonists/space_dragon/space_dragon.dm [hunk 1]
```diff
@@ -92,6 +92,7 @@
 		to_chat(owner.current, span_boldwarning("You've failed to summon the rift in a timely manner! You're being pulled back from whence you came!"))
 		destroy_rifts()
 		SEND_SOUND(owner.current, sound('sound/magic/demon_dies.ogg'))
+		owner.current.death(TRUE)
 		QDEL_NULL(owner.current)
```
Comment (JohnFulpWillard) [inline code/modules/antagonists/space_dragon/space_dragon.dm:96]: Why are we deleting user? Should just kill it I think

Comment (JohnFulpWillard) [reply code/modules/antagonists/space_dragon/space_dragon.dm:96]: https://www.youtube.com/watch?v=dPriJzclQK8

File: code/modules/antagonists/space_dragon/space_dragon.dm [hunk 2]
```diff
@@ -92,6 +92,7 @@
 		to_chat(owner.current, span_boldwarning("You've failed to summon the rift in a timely manner! You're being pulled back from whence you came!"))
 		destroy_rifts()
 		SEND_SOUND(owner.current, sound('sound/magic/demon_dies.ogg'))
+		owner.current.death(TRUE)
```
Comment (Mothblocks) [inline code/modules/antagonists/space_dragon/space_dragon.dm:95]: Named parameter for TRUE

**Review by Mothblocks — APPROVED** [review]
Test is fine

File: code/modules/antagonists/space_dragon/space_dragon.dm (hunk 2 above)
Comment (Mothblocks) [reply code/modules/antagonists/space_dragon/space_dragon.dm:95]: ugh. i remember this

Comment (Mothblocks) [reply code/modules/antagonists/space_dragon/space_dragon.dm:95]:
```suggestion
		owner.current.death(/* gibbed = */ TRUE)
```

this is what i did in one place

**Pull request merged**